plugins/Actions/Columns/ActionType.php

```diff
@@ -40,12 +32,31 @@ class ActionType extends ActionDimension
 
     public function __construct()
     {
-        $this->acceptValues = sprintf('A type of action, such as: %s', implode(', ', $this->types));
+        $this->acceptValues = sprintf('A type of action, such as: %s', implode(', ', $this->getEnumColumnValues()));
     }
 
     public function getEnumColumnValues()
     {
-        return $this->types;
+        $types = [];
+        /**
+         * Triggered to determine the available action types
+         *
+         * Plugin can use this event to add their own action types, so they are available in segmentation
+         * The array maps internal ids to readable action type names used in visitor details
+         *
+         * **Example**
+         *
+         * public function registerActionTypes(&$types)
+         * {
+         *     $types += array(
+         *         76 => 'media_play'
+         *      );
+         * }
+         *
+         * @param array $types
+         */
+        Piwik::postEvent('Actions.registerActionType', [&$types]);
```

Comment:
I wonder if it would make sense to define it in an `array` like this: 
```php
array(
array('id' => 76, 'name' => 'media_play')
)
```
The problem is that those action types are based on numbers and are totally "made up" by the plugin. So it could happen that there are two plugins defining the same action ID. By having it in an array like above we could iterate over it, and throw an exception as soon as two plugins have the same ID (maybe only during dev mode?). The good thing be also that the structure is more extensible in the future and could add possibly add more attributes but might not be needed. 

Alternatively, because there is already a method `getAllActions()` maybe we could think about adding a method `getType()` and maybe `getTypeName()` to the `Piwik\Tracker\Action` and each action could return the type they define so we wouldn't need to add a new event (by default nothing would be returned)? I reckon it be good to have all such related code in one class and not needing an event.

In case we keep the event, should we rather use `Actions.addActionTypes` for consistent naming with other events?

I have no clear preference here but be good if we could catch it in case two plugins define the same type number.

Reply:
Couldn't find a class that defines multiple types. Is it in core?

Reply:
> I don't think we are using this anywhere

Its more a theoretical use case, not anything that is in use in any of our plugins afaik

Reply:
👍  As it is mainly used by us anyway it is not needed to use action classes

Reply:
If updated the PR. It still uses event but now check if a action with same id is defined multiple times